src/main/java/Game.java

```diff
@@ -162,4 +172,134 @@ private Position getPositionFromUser() {
         }
     }
 
+    @NotNull
+    private Player getTheWinner() {
+        for (Map.Entry<Position, Cell> entry : board) {
+            Position currentPosition = entry.getKey();
+            checkFreedomLine(currentPosition);
+        }
+        if(blackScore.getNumberOfFreedomLines() > whiteScore.getNumberOfFreedomLines()) {
+            return blackPlayer;
+        }
+        return whitePlayer;
+    }
+
+
+    private void checkFreedomLine(@NotNull Position position) {
+        Stone.Color color = board.getStone(position).getColor();
```

Comment:
This works when the board is full, but if a cell is free it returns a `NullPointerException`: should add a check whether the cell is occupied

Reply:
You are right